### ISSUE TITLE
dplyr 1.0 full support

### ISSUE THREAD

**Issue by gergness** [gergness/srvyr#79]
Jotting down a few things I'll want to work on when I actually support 1.0:

- Take out delayed s3 registration for `pull()`
- Support `summarise()` .groups argument
- Investigate how the new multiple-value returns are allowed from summarise to see if srvyr's methods can be simplified and/or made to better match dplyr's evolution

**Comment by gergness** [comment]
From #80 

- The function transmute doesn't work.
- Alternatively, the new argument .keep in mutate isn't supported.

**Comment by acircleda** [comment]
Thanks for the package and keeping it updated with 1.0! So much easier to use `dplyr` functions than in the `survey` package.

**Comment by theiostream** [comment]
Hi there! Just noting that `survey_count()` should pass `.add`, rather than `add`, to dplyr's `group_by`.

**Comment by palmaresk8** [comment]
Yes, when using `group_by` we get the following warning: 
`Warning message:
The `add` argument of `group_by()` is deprecated as of dplyr 1.0.0.
Please use the `.add` argument instead.`

**Comment by labreumaia** [comment]
Thanks for the great package. Do you intend to also make srvyr compatible with rowwise functions? This would make it much easier for us modeling with svyglm.

**Comment by gergness** [comment]
@labreumaia - it's a possibility, but I'm not sure I understand. Can you explain your usecase? Why would rowwise help with svyglm?

**Comment by labreumaia** [comment]
It would facilitate running model in different subgroups – e.g., different
years, in case of a time series, different states, etc. It is basically the
same idea of grou_by, but row-based, rather than column based.
-- Lucas de Abreu Maia
Pre-doctoral fellow | Massachusetts Institute of Technology
Ph.D. candidate | Department of Political Science | University of
California, San Diego
lucasamaia.com

**Comment by gergness** [comment]
Hm, but how do you run a model on a single row? Are you assuming you’ve nested first?

In that case maybe srvyr could help with the nesting but I think the outer layer is a data.frame with a column of tbl_svy, so you’d use traditional dplyr commands.

**Comment by labreumaia** [comment]
Sorry, I meant rowwise syntax, which includes functions such as nest_by. I recommend having a look at vignette("rowwise"), if you haven't already, especially the modeling subsection: https://cran.r-project.org/web/packages/dplyr/vignettes/rowwise.html. 

I found a way to integrate rowwise programming with srvyr by creating a different tbl_svy object for each subset of the data, e.g.:

```
mods <- my_data %>%
  nest_by(subsetting_var) %>%
  mutate(design = list(as_survey(data),
  mod = list(svyglm(y ~ x, design)))
```

This isn't ideal, because one needs to recreate the tbl_svy objects if one is running different model specifications in different code chunks. I think the ideal syntax would be something like:

```
my_survey <- my_data %>%
  as_survey

mods <- my_survey %>%
  nest_by(subsetting_var) %>%
  mutate(mod = list(svyglm(y ~ x, data)))
```

Hence the survey design is created only once.

**Comment by gergness** [comment]
For people interested in this thread, the `rowwise` branch (install via `remotes::install_github("gergness/srvyr", ref = "across")`) is starting to take form.

Notably it includes `nest_by()`:
``` r
library(srvyr)
library(survey)

data(api)

dstrata <- apistrat %>%
  as_survey_design(strata = stype, weights = pw)

out <- dstrata %>%
  nest_by(stype) %>%
  mutate(model = list(svyglm(api00~ell+meals+mobility, design = data)))

summary(out$model[[1]])
#> 
#> Call:
#> svyglm(formula = api00 ~ ell + meals + mobility, design = data)
#> 
#> Survey design:
#> Called via srvyr
#> 
#> Coefficients:
#>             Estimate Std. Error t value Pr(>|t|)    
#> (Intercept) 860.5911    11.3699  75.690   <2e-16 ***
#> ell          -0.4083     0.4144  -0.985    0.327    
#> meals        -3.4820     0.3096 -11.246   <2e-16 ***
#> mobility      0.2386     0.2993   0.797    0.427    
#> ---
#> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
#> 
#> (Dispersion parameter for gaussian family taken to be 3202.722)
#> 
#> Number of Fisher Scoring iterations: 2
```


And `dplyr::across()` is supported

``` r
library(srvyr)
library(survey)

data(api)

dstrata <- apistrat %>%
  as_survey_design(strata = stype, weights = pw)

dstrata %>% 
  group_by(stype) %>%
  summarize(across(starts_with("api"), ~survey_mean(., vartype = "ci")))
#> # A tibble: 3 x 10
#>   stype api00 api00_low api00_upp api99 api99_low api99_upp api.stu api.stu_low
#>   <fct> <dbl>     <dbl>     <dbl> <dbl>     <dbl>     <dbl>   <dbl>       <dbl>
#> 1 E      674.      650.      699.  636.      610.      662.    355.        327.
#> 2 H      626.      595.      656.  617.      586.      649.   1071.        906.
#> 3 M      637.      604.      669.  610.      577.      643.    696.        597.
#> # … with 1 more variable: api.stu_upp <dbl>
```

There were a lot of changes under the hood, but I think it's mostly working? Would be happy to get feedback. 

The only breaking change I know of is that the order of computation is changed if an argument of `summarize` matches an existing variable. This code works in the CRAN release, but errors in this branch. You can fix by using a different name in the output of `summarize`.


``` r
library(srvyr)
library(survey)

data(api)

dstrata <- apistrat %>%
  as_survey_design(strata = stype, weights = pw)

dstrata %>%
  summarize(
    api99 = survey_mean(api99), 
    api_diff = survey_mean(api00 - api99)
  )
#> Error: Problem with `summarise()` input `api_diff`.
#> x invalid type (list) for variable '__SRVYR_TEMP_VAR__'
#> ℹ Input `api_diff` is `survey_mean(api00 - api99)`.
#> Backtrace:
#>      █
#>   1. └─`%>%`(...)
#>   2.   ├─base::withVisible(eval(quote(`_fseq`(`_lhs`)), env, env))
#>   3.   └─base::eval(quote(`_fseq`(`_lhs`)), env, env)
#>   4.     └─base::eval(quote(`_fseq`(`_lhs`)), env, env)
#>   5.       └─`_fseq`(`_lhs`)
#>   6.         └─magrittr::freduce(value, `_function_list`)
#>   7.           ├─base::withVisible(function_list[[k]](value))
#>   8.           └─function_list[[k]](value)
#>   9.             ├─dplyr::summarize(...)
#>  10.             └─srvyr:::summarise.tbl_svy(...)
#>  11.               ├─dplyr::summarise(.data$variables, ..., .groups = .groups) /Users/gregfreedmanellis/srvyr/R/summarise.r:10:2
#>  12.               └─dplyr:::summarise.data.frame(.data$variables, ..., .groups = .groups)
#>  13.                 └─dplyr:::summarise_cols(.data, ...)
#> <parent: error/simpleError>
#> Backtrace:
#> █
```

**Comment by gergness** [comment]
closed via #89
